### PR TITLE
4.x request referer

### LINKED ISSUE #12402: $this->referer() returns '/' if $ref == empty. Expected a null/empty return value
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [x ] feature-discussion (RFC)

* CakePHP Version: 3.5.1

### What you did
I needed to check the referer on a controller for some params, so i used $this->referer(). This works when there is a referer, but when the page is called directly, the call to referer() returns '/'. 

 ```
   public function referer($local = false)
    {
        $ref = $this->getEnv('HTTP_REFERER');

        $base = Configure::read('App.fullBaseUrl') . $this->webroot;
        if (!empty($ref) && !empty($base)) {
            if ($local && strpos($ref, $base) === 0) {
                $ref = substr($ref, strlen($base));
                if (!strlen($ref)) {
                    $ref = '/';
                }
                if ($ref[0] !== '/') {
                    $ref = '/' . $ref;
                }

                return $ref;
            }
            if (!$local) {
                return $ref;
            }
        }

        **return '/';**
    }

```

### What you expected to happen
I was expecting an empty return value for an empty referer.

### PR DESCRIPTION
**Pull request by ADmad**
Closes #12402.

### CODE REVIEW

File: src/Http/ServerRequest.php [hunk 1]
```diff
@@ -589,7 +589,7 @@ public function referer(bool $local = false): string
             }
         }
 
-        return '/';
+        return null;
```
Comment (markstory) [inline src/Http/ServerRequest.php]: What do browsers do if you set `Location: ` header to an empty value?

Comment (ADmad) [reply src/Http/ServerRequest.php]: I made a simple test using `<?php header('Location:');` and FF seem to redirect to same page while chrome does nothing.

I think it's upto developer to check the return value if he/she want's to use it for redirection. We have `Controller::redirection()` which returns a default value if no referrer is found for such cases.

Comment (markstory) [reply src/Http/ServerRequest.php]: Good point about Controller::redirect() being able to handle higher level use cases.

**Comment by markstory** [comment]
I will get this merged when I get the migration guide updated.

**Pull request merged**